torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3340,6 +3340,28 @@ std::tuple<Tensor, Tensor> polar_backward(
   return std::make_tuple(grad_abs, grad_angle);
 }
 
+Tensor i1_backward(const Tensor& grad, const Tensor& self, const Tensor& result) {
+  return AT_DISPATCH_FLOATING_TYPES_AND2(
+      at::kHalf, at::kBFloat16, self.scalar_type(), "i1_backward", [&]() {
+        auto eps = std::numeric_limits<scalar_t>::epsilon();
```

Comment:
This is going to need a comment explaining the approach

Reply:
Updated.